kroxylicious-filters/kroxylicious-authorization/src/main/java/io/kroxylicious/filter/authorization/DeleteTopicsEnforcement.java

```
@@ -44,26 +49,59 @@ CompletionStage<RequestFilterResult> onRequest(RequestHeaderData header,
         if (useStates) {
-            actions = operation.actionsOf(request.topics().stream()
-                    .map(t -> {
-                        if (t.name() != null) {
-                            return t.name();
-                        }
-                        else {
-                            throw authorizationFilter.topicIdsNotSupported();
-                        }
-                    }));
+            var partitioned = request.topics().stream().collect(Collectors.partitioningBy(t -> t.name() != null));
+            List<DeleteTopicsRequestData.DeleteTopicState> statesUsingIds = partitioned.get(false);
+            if (statesUsingIds.isEmpty()) {
+                actionsStage = CompletableFuture.completedStage(operation.actionsOf(partitioned.get(true).stream()
+                        .map(DeleteTopicsRequestData.DeleteTopicState::name)));
+                mappingStage = null;
+            }
+            else {
+                mappingStage = new CompletableFuture<>();
+                context.topicNames(statesUsingIds.stream()
+                        .map(DeleteTopicsRequestData.DeleteTopicState::topicId)
+                        .toList()).whenComplete((mapping, error) -> {
+                            // painfully, the stage returned by context.topicNames() cannot be converted to a future
+                            // so we have to make our own
+                            if (error == null) {
+                                mappingStage.complete(mapping);
+                            }
+                            else {
+                                mappingStage.completeExceptionally(error);
+                            }
+                        });
+                CompletionStage<List<Action>> actionsStage1 = mappingStage.thenApply(mapping -> {
+                    Stream<String> names = request.topics().stream()
+                            .map(t -> {
+                                return mapping.topicNames().get(t.topicId());
```

What should the behaviour be when we fail to map an id? Set an unknown server exception for that topic?

---

It would depend on the error. For example we should probably propagate an `UNKNOWN_TOPIC_ID` back to the client. Which makes me realise that we're missing test coverage for cases like that.